src/posix/platform/netif.cpp

```diff
@@ -298,36 +325,81 @@ static uint8_t NetmaskToPrefixLength(const struct sockaddr_in6 *netmask)
 }
 #endif
 
-static void UpdateUnicast(otInstance *aInstance, const otIp6Address &aAddress, uint8_t aPrefixLength, bool aIsAdded)
+#if defined(__linux__)
+static void UpdateUnicastLinux(const otIp6AddressInfo &aAddressInfo, bool aIsAdded)
 {
-    OT_UNUSED_VARIABLE(aInstance);
+    struct rtattr *rta;
 
-    otError error = OT_ERROR_NONE;
+    struct
+    {
+        struct nlmsghdr  nh;
+        struct ifaddrmsg ifa;
+        char             buf[512];
+    } req;
 
-    assert(sInstance == aInstance);
+    memset(&req, 0, sizeof(req));
 
-    VerifyOrExit(sIpFd >= 0, error = OT_ERROR_INVALID_STATE);
+    req.nh.nlmsg_len   = NLMSG_LENGTH(sizeof(struct ifaddrmsg));
+    req.nh.nlmsg_flags = NLM_F_REQUEST | NLM_F_ACK | NLM_F_CREATE | NLM_F_EXCL;
+    req.nh.nlmsg_type  = aIsAdded ? RTM_NEWADDR : RTM_DELADDR;
+    req.nh.nlmsg_pid   = 0;
+    req.nh.nlmsg_seq   = ++sNetlinkSequence;
 
-#if defined(__linux__)
+    req.ifa.ifa_family    = AF_INET6;
+    req.ifa.ifa_prefixlen = aAddressInfo.mPrefixLength;
+    req.ifa.ifa_flags     = aAddressInfo.mIsAnycast ? IFA_F_NODAD : 0;
```

Comment:
Should use `IFA_F_NODAD` for all Thread addresses? I think OpenThread should manage it's addresses by itself. Thus, no DAD is required for these addresses.